src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -161,25 +168,63 @@ internal static BindingDataResolver GetAppSettingResolver(string originalValue,
             return (newAttr, bindingData) => resolvedValue;
         }
 
+        // Run validition. This needs to be run at different stages. 
+        // In general, run as early as possible. If there are { } tokens, then we can't run until runtime.
+        // But if there are no { }, we can run statically. 
+        // If there's no [AutoResolve], [AppSettings], then we can run immediately.
+        private static void RunValidateAttributes(object value, PropertyInfo propInfo)
+        {
+            var attrs = propInfo.GetCustomAttributes<ValidationAttribute>();
+
+            foreach (var attr in attrs)
+            {
+                try
+                {
+                    attr.Validate(value, propInfo.Name);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"'{propInfo.Name}' can't be '{value}': {e.Message}");
```

Comment:
This code enables more than just the existing table name validation code, it can be used by any extension author - unfortunately I think that the broad brush is what we need, as we can't ensure that they will use a [SecureString] attribute correctly.

Not logging the value if also annotated with [AppSetting] might be slightly better.